x-pack/plugins/cloud_security_posture/public/pages/rules/rules_table.tsx

```diff
@@ -37,55 +50,166 @@ export const RulesTable = ({
   };
 
-  const rowProps = (row: CspBenchmarkRule) => ({
-    ['data-test-subj']: TEST_SUBJECTS.getCspBenchmarkRuleTableRowItemTestId(row.metadata.id),
+  const rowProps = (row: CspBenchmarkRulesWithStatus) => ({
+    ['data-test-subj']: TEST_SUBJECTS.getCspBenchmarkRuleTableRowItemTestId(row.metadata?.id),
     style: {
-      background: row.metadata.id === selectedRuleId ? euiTheme.colors.highlight : undefined,
+      background: row.metadata?.id === selectedRuleId ? euiTheme.colors.highlight : undefined,
     },
   });
 
+  const [selectAllRulesThisPage, setSelectAllRulesThisPage] = useState<boolean>(false);
+  const postRequestChangeRulesStatus = useChangeCspRuleStatus();
+
+  const isArraySubset = (
+    smallArr: CspBenchmarkRulesWithStatus[],
+    bigArr: CspBenchmarkRulesWithStatus[]
+  ) => {
+    let i: number = 0;
+    const newSmallArr = smallArr.map((e) => e.metadata);
+    const newBigArr = bigArr.map((e) => e.metadata);
+
+    while (i < newSmallArr.length) {
+      if (!newBigArr.includes(newSmallArr[i])) return false;
+      i++;
+    }
+    return true;
+  };
+
+  useEffect(() => {
+    if (selectedRules.length >= items.length) setSelectAllRulesThisPage(true);
+  }, [items.length, selectedRules.length]);
+
+  const columns = useMemo(
+    () =>
+      getColumns({
+        setSelectedRuleId,
+        refetchStatus,
+        postRequestChangeRulesStatus,
+        selectedRules,
+        setSelectedRules,
+        items,
+        setSelectAllRulesThisPage,
+        selectAllRulesThisPage,
+        isArraySubset,
+      }),
+    [
+      setSelectedRuleId,
+      refetchStatus,
+      postRequestChangeRulesStatus,
+      selectedRules,
+      setSelectedRules,
+      items,
+      setSelectAllRulesThisPage,
+      selectAllRulesThisPage,
+    ]
+  );
+
   return (
-    <EuiBasicTable
-      data-test-subj={TEST_SUBJECTS.CSP_RULES_TABLE}
-      loading={loading}
-      error={error}
-      items={items}
-      columns={columns}
-      pagination={euiPagination}
-      onChange={onTableChange}
-      itemId={(v) => v.metadata.id}
-      rowProps={rowProps}
-    />
+    <>
+      <EuiBasicTable
+        data-test-subj={TEST_SUBJECTS.CSP_RULES_TABLE}
+        loading={loading}
+        error={error}
+        items={items}
+        columns={columns}
+        pagination={euiPagination}
+        onChange={onTableChange}
+        itemId={(v) => v.metadata.id}
+        rowProps={rowProps}
+      />
+    </>
   );
 };
 
-type GetColumnProps = Pick<RulesTableProps, 'setSelectedRuleId'>;
-
 const getColumns = ({
   setSelectedRuleId,
-}: GetColumnProps): Array<EuiTableFieldDataColumnType<CspBenchmarkRule>> => [
+  refetchStatus,
+  postRequestChangeRulesStatus,
+  selectedRules,
+  setSelectedRules,
+  items,
+  setSelectAllRulesThisPage,
+  selectAllRulesThisPage,
+  isArraySubset,
+}: GetColumnProps & any): Array<EuiTableFieldDataColumnType<CspBenchmarkRulesWithStatus>> => [
+  {
+    field: 'action',
```

Comment:
correct me if im wrong here but did you basically tried to implement the selection mechanism for this table? i think it should come out of the box for EUI table by using the `isSelectable={true}` prop.

Please read the API here: https://eui.elastic.co/#/tabular-content/tables#adding-selection-to-a-table
there are a bunch of controllers as well, i think we should stick to the existing API and not redo it ourselves even if it means the UI is not the exactly the same as figma

Reply:
Originally I have that, But we found an issue with using the isSelectable prop such as causing an issue when implementing the feature for user to choose all rules (across all pages) and after having a discussion with @opauloh , we decided to implement our own